### ISSUE TITLE
Would it be possible to add some way to add platform compile clause (like preprocessor directives)

### ISSUE THREAD

**Issue by cyborgar** [irmen/prog8#30]
I.e I'm working on some code that is almost platform independent (so far). I just have 1 line of CX16 specific code I need to remove to get my code to compile on both platforms.  Basically I'm asking for something like this

```
#ifdef TARGET_C64
  void cx16.screen_set_mode(0)
#endif
```

If I ever consider adding sound this might possible be done with a similar solution just around the %import

```
#ifdef TARGET_C64
  %import c64_sound
#endif
#ifdef TARGET_CX16
  %import cx16_sound
#endif
```

or perhaps some target specific flag like this 

`%import %target%_sound`

might be possible instead

**Comment by cyborgar** [comment]
Just in case you are curious, I have added you to the code repo.

I intend to just make this open when it's a bit more complete. Though I don't expect a huge amount of interest :-)

**Comment by irmen** [comment]
I don't intend to use a text based preprocessing mechanism.
This also means that as a directive isn't a statement, you can't stick them inside other statements like an if (which would be meta-programming).
What I *am* doing is making a command line option that let you tell the compiler where to look for imported modules. So like the -target option you can add a second option that points the compiler to a different location as desired. You can then create two versions of a certain module ("systemspecific.p8" perhaps?) one in each location.
I think that should work

P.S. I can't look into your repo because it's private

**Comment by irmen** [comment]
@cyborgar that project you made looks AMAZING !   What an awesome job!  How is your experience with using prog8 for it?

**Comment by cyborgar** [comment]
> What I _am_ doing is making a command line option that let you tell the compiler where to look for imported modules. So like the -target option you can add a second option that points the compiler to a different location as desired. You can then create two versions of a certain module ("systemspecific.p8" perhaps?) one in each location.
> I think that should work

That would work. And it wouldn't be hard split out the difference in separate modules even if the modules will be trivial.
BTW, It would also be easy to just make a petaxion64.p8 "front" do the variant importing there. But it's sort of nice to have duplications limited to the absolute minimum

And thanks for liking the project. I'll upload it when I have a some "game over" handling. 

It's been very fun coding in prog8 so far. And working with the more limited platform. I have only vauge feelings for what results in effecient code so far and I'm experimenting as I work. And figuring how to make the game loop is new to me. The only games I have implemented before are snake and tetris (and that was 30 years ago).

BTW, optimization for speed only became an issue today when I compiled for C64 for the first time. There I can clearly see that there is a speed difference between the C64 and XC16.
